idaes/core/util/scaling.py

```diff
@@ -611,6 +646,60 @@ def extreme_jacobian_entries(
                 el.append((e, c, v))
     return el
 
+def extreme_jacobian_rows(
+        m=None, scaled=True, large=1e4, small=1e-4, jac=None, nlp=None):
+    """
+    Show very large and very small Jacobian rows. Typically indicates a badly-
+    scaled constraint.
+
+    Args:
+        m: model
+        scaled: if true use scaled Jacobian
+        large: >= to this value is consdered large
+        small: <= to this is consdered small
+
+    Returns:
+        (list of tuples), Jacobian entry, Constraint
+    """
+    if jac is None or nlp is None:
+        jac, nlp = get_jacobian(m, scaled)
+    el = []
+    for i, c in enumerate(nlp.clist):
+        norm = 0
+        for j in jac[i].indices:
+            norm += jac[i, j]**2
```

Comment:
For example, add a comment that this is computing an L2 norm (which helps explain the **2)